release-notes/7.0/supported-os.json

```diff
@@ -0,0 +1,64 @@
+{
+  "channel-version": "7.0",
+  "last-updated": "2024-02-28",
+  "glibc": [
```

Comment:
I tried to make sense from https://github.com/Falco20019/core/blob/extract-release-json/release-notes/7.0/supported-os.md#libc-compatibility but am not completely sure I got it right.

* `CentOS 7` only support `x64`, so that sounds plausible
* `Ubuntu 18.04` supports `x64, `arm64` and `arm`, but only `arm` is listed for `Ubuntu 18.04` whereas `arm64` is the only listed for `Ubuntu 16.04`
* `arm64` seems to have a higher min-version for `.NET 7.0.3 and earlier` (which OS does this affect?)
* `Alpine 3.15` only supports `x64` and `arm64` through `musl`, but thre is also `arm` support listed above

@richlander Is `glibc` and `musl` required version tightly linked to the OS and architecture and is the OS cycle the min version including all following if not defined otherwise? In that case, I would rename `cycle` to `min-cycle` or change `cycle` to an array like with `supported-cycles`.

Reply:
I don't think musl vs glibc makes sense as a top-level notion. It is a very secondary concern for most users, particularly as it relates to EOL information.

I don't think it makes sense to list the OSes we use in our build system in this file. That's an implementation detail. The contract is the min versions that result from those choices. That does make sense to list, same as we do in the md file.

I think a `libc` property could make sense for Linux distros, however, it doesn't add much since we currently only support one muslc distro and I don't see that changing soon.

Reply:
I can make a proposal on this.

Reply:
Proposal for that section. The OS content would come later. I removed it for brevity.

```json
{
  "channel-version": "6.0",
  "last-updated": "2024-02-28",
  "glibc": [
    {
      "architecture" : "x64",
      "version": "2.17",
      "source": "CentOS 7"
    },
    {
      "architecture" : "arm64",
      "version": "2.23",
      "source": "Ubuntu 16.04"
    },
    {
      "architecture" : "arm32",
      "version": "2.23",
      "source": "Ubuntu 16.04"
    }
  ],
  "musl" : [
    {
      "architecture" : "x64",
      "version": "1.2.2",
      "source": "Alpine 3.13"
    },
    {
      "architecture" : "arm64",
      "version": "1.2.2",
      "source": "Alpine 3.13"
    },
    {
      "architecture" : "arm32",
      "version": "1.2.2",
      "source": "Alpine 3.13"
    }
  ]
}
```

Reply:
I think using strings as `source` might make it harder to work with in a parsable form (compared to having it split up into os-name and os-cycle). I also find it strange to duplicate full items in the `musl` case, as it might introduce the need for users to compare and join those manually.

Reply:
The strings are solely there to enable printing the following text.

<img width="388" alt="image" src="https://github.com/dotnet/core/assets/2608468/a88ea821-018e-4dfa-bd02-55d539240ac9">

https://github.com/dotnet/core/blob/main/release-notes/8.0/supported-os.md#libc-compatibility

I don't have another use case in mind where the "source" values would be needed. Do you? There is no user action to take as a result of that information (other than not running .NET 6 on Alpine 3.12, for example).

Reply:
The duplication is unfortunate, however, I'm not sure how to solve that w/o adding more complexity to the format. This is effectively 1NF in database schema. It works. As demonstrated, the `glibc` builds require this approach. It' possible that we'll never need this per-architecture description again, but that's not a great bet to make.

Reply:
We could do this:

```json
{
  "channel-version": "6.0",
  "last-updated": "2024-02-28",
  "glibc": [
    {
      "architecture" : "x64",
      "version": "2.17",
      "source": "CentOS 7"
    },
    {
      "architecture" : "arm64",
      "version": "2.23",
      "source": "Ubuntu 16.04"
    },
    {
      "architecture" : "arm32",
      "version": "2.23",
      "source": "Ubuntu 16.04"
    }
  ],
  "musl" : [
    {
      "version": "1.2.2",
      "source": "Alpine 3.13"
    }
  ]
}
```

That would be intended to mean "all architecures" for musl.